### PR TITLE
Dockerfiles: extract CI dockerfile

### PR DESCRIPTION
**Pull request by fromanirh**
Split the Dockerfiles to build the index image to make integration
with CI easier.

Signed-off-by: Francesco Romani <fromani@redhat.com>

### CODE REVIEW

**Comment by fromanirh** [comment]
@cynepco3hahue @yanirq

File: openshift-ci/Dockerfile.index.ci [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM quay.io/operator-framework/upstream-opm-builder
```
Comment (yanirq) [inline openshift-ci/Dockerfile.index.ci:1]: where is this Dockerfile.index.ci image used ?

Comment (fromanirh) [reply openshift-ci/Dockerfile.index.ci:1]: Here: https://github.com/openshift/release/pull/10474/commits/d17a2bf097412800daaa6ddeee44f9f9e3d00868#diff-fb490d3ebe4d085e38b1a4faf94429adR46

Comment (fromanirh) [reply openshift-ci/Dockerfile.index.ci:1]: The issue here is the nested path. the ".ci" variants have a flatter path which save us quit a lot of complications in openshift CI.

**Pull request merged**